server/mdm/apple/apple_mdm.go

```diff
@@ -212,23 +214,34 @@ func (d *DEPService) RegisterProfileWithAppleDEPServer(ctx context.Context, team
 	jsonProf.AwaitDeviceConfigured = true
 
 	depClient := NewDEPClient(d.depStorage, d.ds, d.logger)
-	res, err := depClient.DefineProfile(ctx, DEPName, &jsonProf)
+	// Get all relevant org names
+	var tmID *uint
+	if team != nil {
+		tmID = &team.ID
+	}
+	orgNames, err := d.ds.GetABMTokenOrgNamesForTeam(ctx, tmID)
 	if err != nil {
-		return ctxerr.Wrap(ctx, err, "apple POST /profile request failed")
+		return ctxerr.Wrap(ctx, err, "getting org names for team to register profile")
 	}
 
-	if setupAsst != nil {
-		setupAsst.ProfileUUID = res.ProfileUUID
-		if err := d.ds.SetMDMAppleSetupAssistantProfileUUID(ctx, setupAsst.TeamID, res.ProfileUUID); err != nil {
-			return ctxerr.Wrap(ctx, err, "save setup assistant profile UUID")
-		}
-	} else {
-		var tmID *uint
-		if team != nil {
-			tmID = &team.ID
+	if len(orgNames) == 0 {
+		orgNames = append(orgNames, DEPName)
+	}
+	for _, orgName := range orgNames {
+		res, err := depClient.DefineProfile(ctx, orgName, &jsonProf)
+		if err != nil {
+			return ctxerr.Wrap(ctx, err, "apple POST /profile request failed")
 		}
-		if err := d.ds.SetMDMAppleDefaultSetupAssistantProfileUUID(ctx, tmID, res.ProfileUUID); err != nil {
-			return ctxerr.Wrap(ctx, err, "save default setup assistant profile UUID")
+
+		if setupAsst != nil {
+			setupAsst.ProfileUUID = res.ProfileUUID
+			if err := d.ds.SetMDMAppleSetupAssistantProfileUUID(ctx, setupAsst.TeamID, res.ProfileUUID); err != nil {
```

Comment:
Does that work? Aren't we overwriting the setup assistant for a given team with whatever ProfileUUID was the latest one registered with Apple?

I think we missed that when specing the story, but the setup assistant table is 1:1 with a team id:

```
  UNIQUE KEY `idx_mdm_setup_assistant_global_or_team_id` (`global_or_team_id`),
```

and only a single profile_uuid associated with it (1:1 with that uuid), and now I think we'd need a setup assistant to have a 1:many relationship with profile_uuids, one for each abm token? @roperzh to double-check my understanding.

Unless the `profile_uuid` is the same as long as the profile is identical (which I vaguely recall might be the case, it's not so much a uuid but a hash?), but this would be relying on an undocumented internal detail of a third-party...

Reply:
oof yes, I think you're right on both things, profile UUID is the same but it's not documented... so we probably need to add 1:many

Reply:
also, I think the same UUID behavior might only be within the same org, not sure if you're using different ABM tokens.

Reply:
Yeah, we've never been able to verify that up to now, only within an org, good point.